### PR TITLE
[15.0][IMP] sale_order_secondary_unit: Compatibility with product variants

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT40323

ping  @carlosdauden @chienandalu 

Depends on:

- [x] https://github.com/OCA/product-attribute/pull/1246

### CODE REVIEW

**Comment by sergio-teruel** [comment]
Hi  @rousseldenis , can you take a look? I have used the test-requirements.txt to import the PR.
Thanks...

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+odoo-addon-product_secondary_unit @ git+https://github.com/OCA/product-attribute.git/refs/pull/1246/head#subdirectory=setup/product_secondary_unit
```
Comment (chienandalu) [inline test-requirements.txt]:
```suggestion
odoo-addon-product_secondary_unit @ git+https://github.com/OCA/product-attribute.git@refs/pull/1246/head#subdirectory=setup/product_secondary_unit
```

Comment (sergio-teruel) [reply test-requirements.txt]: upps!!! The doc is wrong.
`odoo-addon-<module_name> @ git+https://github.com/OCA/<repository>.git/refs/pull/<PR_number>/head#subdirectory=setup/<module_name>`

I will try it!! Thanks @chienandalu

Comment (pedrobaeza) [reply test-requirements.txt]: I have just fixed the documentation. The example were good though.

**Review by chienandalu — APPROVED** [review]
Ok :+1:

**Comment by sergio-teruel** [comment]
New commit to improve the sale order line kanban view https://github.com/OCA/sale-workflow/pull/2331/commits/50473cf480ef4dabd52f49fdbf560275fd859900

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major

**Comment by sergio-teruel** [comment]
Tests fixed!!

**Review by carlosdauden — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@pedrobaeza @sergio-teruel why the field `secondary_uom_unit_price` was deleted from the view yet kept in the model?

https://github.com/OCA/sale-workflow/pull/2331/files#diff-f206d07f7cdb9ec1f638ab75260d3e525b36c7f7dc4590562c6930540f51beb3L28

File: sale_order_secondary_unit/views/sale_order_views.xml [hunk 2]
```diff
@@ -12,20 +12,25 @@
...
-                    optional="show"
-                />
-                <field
-                    name="secondary_uom_id"
-                    class="oe_inline"
-                    domain="[('product_tmpl_id.product_variant_ids', 'in', [product_id])]"
-                    attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
-                    optional="show"
-                />
-                <field name="secondary_uom_unit_price" readonly="1" optional="show" />
```
Comment (alexey-pelykh) [inline sale_order_secondary_unit/views/sale_order_views.xml:28]: @pedrobaeza and this - why keep the field in model if not used anywhere?

Comment (pedrobaeza) [reply sale_order_secondary_unit/views/sale_order_views.xml:28]: I'm not sure. I hope @sergio-teruel can tell.

Comment (sergio-teruel) [reply sale_order_secondary_unit/views/sale_order_views.xml:28]: This field was added in the form view, this view normally is not used to create sale lines. And I saw that is not defined on tree view.. but I do not remember de author of it, mybe he can tell us more about these need..

Comment (alexey-pelykh) [reply sale_order_secondary_unit/views/sale_order_views.xml:28]: But the field was readonly so I assume that was informational field intended for viewing? I know that sale order line form is rarely used, still I don't think it was worth removing it

**Pull request merged**